R/force_sorted.R

```diff
@@ -0,0 +1,51 @@
+#' @title Forcing a numeric vector into a monotonously increasing sequence.
+#'
+#' @description
+#' This function performs interpolation on the non-increasing parts of an
+#' input vector to ensure its values are continuously increasing.
+#'
+#' @param x `numeric` vector.
+#'
+#' @return A vector with continuously increasing values.
+#'
+#' @note
+#' NA values will not be replaced.
+#'
+#' @examples
+#' x <- c(NA, NA, NA, 1.2, 1.1, 1.14, 1.2, 1.3, NA, 1.04, 1.4, 1.6, NA, NA)
+#' sorted_rtime <- force_sorted(x)
+#' is.unsorted(x, na.rm = TRUE)
+#'
+#' @export
+#'
+#' @rdname force_sorted
+force_sorted <- function(x){
+    # Select only the non-NA values
+    if (!is.numeric(x) && !is.integer(x))
+        stop("'x' needs to be numeric or integer")
+    nna_idx <- which(!is.na(x))
+    vec_temp <- x[nna_idx]
+
+    while (any(diff(vec_temp) < 0)) {
+        idx <- which.max(diff(vec_temp) < 0)
+        # Find next biggest value
+        next_idx <- which(vec_temp > vec_temp[idx])[1L]
+
+        if (is.na(next_idx)){
+            l <- idx:length(vec_temp)
+            vec_temp[l] <- seq(vec_temp[idx], by = 0.000001,
+                               length.out = length(l))
+            warning("Found decreasing values at the end of vector, ",
+                    "interpolation not possible. Replacing values. See help ",
+                    "for more details")
+            break
```

Comment:
What is the output supposed to be after the warning and the `break`? There's not detail in the manual, even though the message tells me to read it.

Reply:
Also, I don't get the warning message. It claims that interpolation isn't possible, even though the output is sorted despite the warning message:

```
> is.unsorted(force_sorted(c(1, 2, 1.5, 2)))
[1] FALSE
Warning message:
In force_sorted(c(1, 2, 1.5, 2)) :
  Found decreasing values at the end of vector, interpolation not possible. Replacing values. See help for more details
```

Reply:
And I see that the `by` is hard-coded to 0.000001. Is there a justification for that value? It could be a function argument?

Reply:
Ah yes I will rewrite the warning to be more clear. 
Even if the interpolation for the last value is not possible the values are increased by 0.0001.
 It make sense for our case but I guess for some user it is not the case, so as you said the `by = 0.0001` could be a function argument !